### PR TITLE
Add docs for storage mapping

### PR DESCRIPTION
**Pull request by oruebel**
This PR adds docs similar to https://nwb-storage.readthedocs.io/en/latest/storage_hdf5.html to describe how elements of the HDMF specification language are mapped to storage in Zarr. This document also describes the implementation of links and object references on the storage side, since links and references are not natively supported by Zarr.

### CODE REVIEW

**Comment by oruebel** [comment]
@alejoe91  this PR may also be of interest for you if you would like to better understand how hdmf-zarr maps NWB types to storage.

File: docs/source/storage.rst [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+Similar to soft links in a file system, a link is an object in a Group that links to another Group or Dataset,
+either within the same Zarr file or another external Zarr file. Links and reference are not natively supported by
+Zarr but are implemented in :py:class:`~hdmf_zarr.backend.ZarrIO` in an OS independent fashion using the ``zarr_link``
+reserved attribute (see :py:attr:`~hdmf_zarr.backend.ZarrIO.__reserve_attribute`) to store a list of dicts serialized
+as JSON. Each dict (i.e., element) in the list defines a link, with each dict containing the following keys:
+
+* ``name`` : Name of the link
+* ``source`` : Relative path to the root of the Zarr file containing the linked object. For links
+  pointing to an object within the same Zarr file, the value of source will be ``"."``. For external
+  links that point ot object in another Zarr file, the value of source will be the  path to
```
Comment (bendichter) [inline docs/source/storage.rst]:
```suggestion
  links that point to object in another Zarr file, the value of source will be the path to
```

File: docs/source/storage.rst [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        },
+        "zarr_dtype": "object"
+    }
+
+.. _sec-zarr-storage-references-region:
+
+Region references
+-----------------
+
+Region references, are similar to object references, but instead of references other Datasets or Groups,
```
Comment (bendichter) [inline docs/source/storage.rst]:
```suggestion
Region references are similar to object references, but instead of references other Datasets or Groups,
```

**Comment by alejoe91** [comment]
> @alejoe91 this PR may also be of interest for you if you would like to better understand how hdmf-zarr maps NWB types to storage.

Thank you @oruebel 

Will test this first thing next week!

**Comment by oruebel** [comment]
@mavaylon1 can you review this PR. This PR only adds documentation but it would be nice to merge it before the release.

**Review by mavaylon1 — APPROVED** [review]

**Pull request merged**